app/src/main/java/com/zulip/android/activities/LoginActivity.java

```diff
@@ -222,6 +225,13 @@ private void showBackends(String httpScheme, String serverURL) {
                     @Override
                     public void onSuccess(Call<ZulipBackendResponse> call, Response<ZulipBackendResponse> response) {
                         if (response.body().isPassword()) {
+
```

Comment:
This hiding of keyboard should be outside isPassword if condition, it's not necessary every server supports EmailBackend!